main/php_variables.c

```diff
@@ -114,6 +114,7 @@ PHPAPI void php_register_variable_ex(const char *var_name, zval *val, zval *trac
 	size_t var_len, index_len;
 	zval gpc_element, *gpc_element_p;
 	bool is_array = 0;
+	bool merge_params = 0;
```

Comment:
IMO, this code does not belong in `php_register_variable_ex`.

Reply:
Is there another place i should look for the implementation of parse_str ?

Reply:
I think we should have specialised function for this as it seems like this use case doesn't exactly fit to the current logic. It's all a bit messy atm.